mongo/integration/initial_dns_seedlist_discovery_test.go

```diff
@@ -62,13 +62,13 @@ func runSeedlistDiscoveryTest(mt *mtest.T, file string) {
 		mt.Skip("skipping to avoid go1.11 problem with multiple strings in one TXT record")
 	}
 
-	cs, err := connstring.Parse(test.URI)
+	cs, err := connstring.ParseAndValidate(test.URI)
 	if test.Error {
 		assert.NotNil(mt, err, "expected URI parsing error, got nil")
 		return
 	}
 	// the resolved connstring may not have valid credentials
-	if err != nil && err.Error() == "error parsing uri: authsource without username is invalid" {
+	if err != nil && err.Error() == "error validating uri: authsource without username is invalid" {
```

Comment:
You can change line 65 to just use connstring.Parse and remove this if statement

Reply:
Oh nice! Just pushed that change.